sherpa/astro/xspec/tests/test_xspec.py

```diff
@@ -805,3 +816,37 @@ def test_apec_redshift_parameter_is_case_insensitive():
     mdl.REDshift = 0.2
     assert mdl.redshift.val == 0.2
     assert mdl.Redshift.val == 0.2
+
+
+@requires_xspec
+def get_xspec_models_with_exp_cache():
+    """Return the XSPEC models that use the modelCacher1d_exp cache.
+
+    Currently, that is: Models that inherit from XSAbsorptionModel.
+    """
+    models = get_xspec_models()
+    return [m for m in models if issubclass(m, XSAbsorptionModel)]
+
+
+@requires_xspec
+@pytest.mark.parametrize("modelcls", get_xspec_models_with_exp_cache())
+def test_models_with_exp_cache(modelcls):
+    '''Tests that models with the modelCacher1d_exp work correctly.
+
+    Unlike modelCacher1d, does not evaluate the model for every
+    parameter value once, but modelCacher1d_exp assumes a certain
+    functional form and calcualtes model values by itself, this this
+    warrantes a test for every model using this cache.
+    '''
+    x = np.arange(0.1, 10, 0.1)
+    xlo = x[:-1]
+    xhi = x[1:]
+    mdl = modelcls()
+    for nh in [1, 0.01, 0.123, 2.3, 11.2]:
+        abs_direct = mdl.calc.__wrapped__(mdl, (nh,), xlo, xhi)
+        abs_cache = mdl.calc((nh,), xlo, xhi)
+        # XSPEC models use float32, so we need to use a relative
+        # tolerance larger than the default
```

Comment:
So, it's a bit more than this, as I see differences across platforms and also we cant guarantee that the model uses float32 - some use float64 (although most of the absorption models do use float32). So I'm not sure a comment here is needed.